### PR TITLE
Swaggerify /register endpoint

### PR DESCRIPTION
**Pull request by Kegsay**
Need to move registration/login/auth sections around once #94 lands.

### CODE REVIEW

File: specification/1-client_server_api.rst [hunk 1]
```diff
@@ -1001,52 +1001,17 @@ member's state, by making a request to
...
-access_token
-  An access token for the new account.
-home_server
-  The hostname of the Home Server on which the account has been registered.
-
-This endpoint may also return the following error codes:
-
-M_USER_IN_USE
-  If the Matrix ID is already in use
-M_EXCLUSIVE
```
Comment (illicitonion) [inline specification/1-client_server_api.rst]: Worth mentioning this still somewhere?

File: specification/1-client_server_api.rst [hunk 2]
```diff
@@ -1001,52 +1001,17 @@ member's state, by making a request to
...
-
-Registering for a user account is done using the request::
-
-  POST $V2PREFIX/register
+Account operations
+------------------
 
+Registration
+~~~~~~~~~~~~
 This API endpoint uses the User-Interactive Authentication API.
```
Comment (illicitonion) [inline specification/1-client_server_api.rst]: Link to the section?

File: specification/1-client_server_api.rst (hunk 1 above)
Comment (Kegsay) [inline specification/1-client_server_api.rst]: Done

File: specification/1-client_server_api.rst (hunk 2 above)
Comment (Kegsay) [inline specification/1-client_server_api.rst]: Done

**Comment by illicitonion** [comment]
LGTM modulo one comment

**Pull request merged**